SConstruct

```diff
@@ -1338,6 +1338,23 @@ env.AlwaysBuild(
             env.PrettyCommand('TIDY', 'src', 'yellow')
         )))
 
+if 'PKG_CONFIG_PATH' in env.Dictionary():
+    pkg_config_path = env['PKG_CONFIG_PATH']
+elif 'PKG_CONFIG' in env.Dictionary() or env.Which('pkg-config'):
+    # https://linux.die.net/man/1/pkg-config the default is libdir/pkgconfig
+    pkg_config_path = os.path.join(env['ROC_SYSTEM_LIBDIR'], 'pkgconfig')
+else:
+    pkg_config_path = None
+if pkg_config_path:
+    pc_target = env.PkgConfig(
+                    build_dir='build',
+                    prefix=GetOption('prefix'),
+                    libdir=env['ROC_SYSTEM_LIBDIR'],
+                    filename='libroc.pc',
+                    version=env['ROC_VERSION'],
+                    dependencies=system_dependencies)
```

Comment:
Just using `system_dependencies` is not correct for two reasons:

* not all of its elements correspond to real libraries, e.g. `stdio` is not a library
* pkg-config files of the real libraries may have different names, e.g. we can have `pulseaudio` in `system_dependencies`, but we should pass `libpulse` to pkg-config

To solve this, I suggest the following approach.

In SConstruct, we have a lot of check like:

```
if 'libuv' in system_dependencies:
    conf = Configure(env, custom_tests=env.CustomTests)

    env.ParsePkgConfig('--cflags --libs libuv')
    ...
```

or:

```
if 'libunwind' in system_dependencies:
    conf = Configure(env, custom_tests=env.CustomTests)

    env.ParsePkgConfig('--cflags --libs libunwind')
    ...
```

These checks already have information what pkg-config dependencies do we need. So let's just modify them so that they accumulate the dependency list for us.

I suggest to replace `env.ParsePkgConfig` (see `site_scons/site_tools/roc/config.py`) with `env.AddPkgConfigDependency` which will be used like this:

```
    env.AddPkgConfigDependency('libunwind', '--cflags --libs')
```

`AddPkgConfigDependency` will do the same as `env.ParsePkgConfig` does currently, but if it succeeds, it will also add the dependency name to `env['PKG_CONFIG_DEPS']`.

And then we can just pass `env['PKG_CONFIG_DEPS']` to `env.GeneratePkgConfig`.

Reply:
thanks for your reply, I am a little bit unclear here.
I added `AddPkgConfigDependency` in `site_scons/site_tools/roc/config.py`
```python
def AddPkgConfigDependency(context, package, flags):
    if context.env.ParsePkgConfig('%s %s' % (package, flags)):
        context.env['PKG_CONFIG_DEPS'].append(package)
        return True

    return False
```
and in SConstruct replace `env.ParsePkgConfig` like this:
```python
if 'libuv' in system_dependencies:
    conf = Configure(env, custom_tests=env.CustomTests)

    conf. AddPkgConfigDependency('libuv', '--cflags --libs')
    ......
    env = conf.Finish()
```
am I understanding correctly?

Reply:
Yeah, something like this, but likely we can remove `ParsePkgConfig` entirely and merge its implementation into `AddPkgConfigDependency`.